File: frontend/Business/Model/GeoCoordinates.py
from pydantic import BaseModel
from typing import List, Optional


class GeoCoordinates(BaseModel):
    id: Optional[int]
    restaurant_id: int
    latitude: float
    longitude: float

    def __init__(self,
                 latitude: float,
                 longitude: float,
                 restaurant_id: int,
                 id: Optional[int] = None):
        super().__init__(
            id=id,
            restaurant_id=restaurant_id,
            latitude=latitude,
            longitude=longitude
        )

    def __str__(self):
        return f"({self.latitude}, {self.longitude})"

File: backend/Domain/Models/GeoCoordinates.cs
using System.Text.Json.Serialization;
using Domain.DataBaseAccess;

namespace Domain.Models;

public class GeoCoordinates(long restaurantId, double latitude, double longitude) : ITableDataBase
{
    public long? Id { get; set; }
    [JsonPropertyName("restaurant_id")] public long RestaurantId { get; init; } = restaurantId;
    public double Latitude { get; init; } = latitude;
    public double Longitude { get; init; } = longitude;

    public override string ToString()
    {
        return $"({Latitude}, {Longitude})";
    }
}